i18n: add ability to omit POST-Creation-Date from being generated

mfn opened the pull request:
Fixes #5887

dereuromark commented on lib/Cake/Console/Command/Task/ExtractTask.php:
```diff
@@ -346,6 +346,10 @@ public function getOptionParser() {
 		))->addOption('extract-core', array(
 			'help' => __d('cake_console', 'Extract messages from the CakePHP core libs.'),
 			'choices' => array('yes', 'no')
+		))->addOption('omit-pot-creation-date', array(
```
`omit-creation-date` would be a little bit shorter maybe.

ADmad replied:
Something shorted like `skip-date` would be better

mfn replied:
Great, changed! I'm notoriously bad at naming things ...

johannesnagl replied:
i like the `skip-date` option naming :)

mfn commented:
I've fixed the tests now too (I don't see how the cachetest relates to my changes; seems like a timing race condition).

Before considering for merge I'd probably clean this up and squash all commits into one.

Q about the tests: wouldn't it be better to set all the default values normally set by `getOptionParser()` into `setUp()` instead of trying to set up the class for each test requiring `->execute()` ?

markstory commented:
Why don't we just remove the date entirely? It doesn't seem like it is that useful

mfn commented:
Well, :+1:  from me. I also don't see its value. I take it given that every project today uses a VCS anyway, but that may just be me.

AD7six commented:
I would also prefer to just get rid of the date, and remove the source of noise.

markstory commented:
Yeah I am for removing the date entirely as well. Do you want to update your pull request @mfn? If not I can remove the troublesome code.

mfn committed:
i18n: Remove POT-Creation-Date

Prevent app locale files getting marked as modified in VCS even if the actual
content didn't change.

Fixes #5887

mfn commented:
Updated/rebased branch!

markstory commented:
Thanks :+1:
